OpenDreamRuntime/DreamConnection.cs

```diff
@@ -210,30 +213,29 @@ public void HandleMsgPromptResponse(MsgPromptResponse message)
 
 
         public void OutputDreamValue(DreamValue value) {
-            if (value.Type == DreamValue.DreamValueType.DreamObject) {
-                DreamObject outputObject = value.GetValueAsDreamObject();
-
+            if (value.Type == DreamValue.DreamValueType.DreamObject && value.TryGetValueAsDreamObject(out var outputObject)) {
                 if (outputObject?.IsSubtypeOf(DreamPath.Sound) == true) {
-                    UInt16 channel = (UInt16)outputObject.GetVariable("channel").GetValueAsInteger();
-                    UInt16 volume = (UInt16)outputObject.GetVariable("volume").GetValueAsInteger();
-                    DreamValue file = outputObject.GetVariable("file");
-
-                    var msg = new MsgSound() {
-                        Channel = channel,
-                        Volume = volume
-                    };
-
-                    if (file.Type == DreamValue.DreamValueType.String || file == DreamValue.Null) {
-                        msg.File = (string)file.Value;
-                    } else if (file.TryGetValueAsDreamResource(out DreamResource resource)) {
-                        msg.File = resource.ResourcePath;
-                    } else {
-                        throw new ArgumentException("Cannot output " + value, nameof(value));
-                    }
+                    if (outputObject.GetVariable("channel").TryGetValueAsInteger(out var channelD)
+                        && outputObject.GetVariable("volume").TryGetValueAsInteger(out var volumeD)) {
```

Comment:
This would lead to a sound not playing if their `channel` or `volume` var was invalid. Some testing in byond shows that they still play, and both have a default of `0`. You can just remove the if and leave it as these two lines:
```
outputObject.GetVariable("channel").TryGetValueAsInteger(out var channelD);
outputObject.GetVariable("volume").TryGetValueAsInteger(out var volumeD);
```
`TryGetValueAsInteger()` defaults to setting the var to 0 if it wasn't an integer.